Feature/styling

abdelp opened the pull request:
This pull request contains all of the necessary files to complete the requirements listed [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896082/)

- Display mode and width for App, Display, and ButtonPanel componets.
- Styling for Display, ButtonPanel, and Buttons

abdelp committed:
Add node-sass npm package

abdelp committed:
Refactor index and App component to use sass

abdelp committed:
Add classNames to the components and display them as flex with 700px width

abdelp committed:
Add Display styling requirements

- gray background
- 100px height
- bold, white colored text
- padding from the edges
- text aligned to the right

abdelp committed:
Add flex-direction to align button groups horizontally

abdelp committed:
Add styling to the buttons

abdelp committed:
Add color to buttons

abdelp committed:
Fix StyleLint offenses

abdelp committed:
Fix ESLint offenses

abdelp committed:
Align display result to the center

abdelp committed:
Update alignment of the app to the center and font-size of buttons

suretrust commented on src/components/Button.js:
```diff
@@ -1,10 +1,10 @@
 
-const Button = ({ buttonName }) => (
-  <button type="button">{buttonName}</button>
+const Button = ({ buttonName, color = 'orange' }) => (
```
- [x] You shouldn't set the value of `color` directly as such. Use `defaultProps` instead. You can read about it [here](https://reactjs.org/docs/typechecking-with-proptypes.html#default-prop-values).

abdelp replied:
changed

abdelp replied:
Refactored

suretrust commented on src/components/Display.js:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropType from 'prop-types';
 
 const Display = ({ result = '0' }) => (
```
Here too.

suretrust commented on src/styles/scss/App.scss:
```diff
@@ -0,0 +1,51 @@
+      font-size: large;
+    }
+
+    .btn-group:last-child .btn:first-child {
+      width: 50%;
+    }
```
- [x] This should be set using `props`, not directly in the CSS.

abdelp replied:
logic moved to the component

suretrust reviewed (changes requested):
Hi @abdelp, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [x] Add props validation to the necessary components.

* [x] The `Button` component should accept `color` and `wide` as an incoming props ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorAndWideProps.png))

* [x] `wide` is boolean, true means that button take 50% of space (used for `0` button) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/WideProps.png))

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

abdelp committed:
Refactor styling

- Add class for wide buttons
- Add prop type validations
- Remove default values from the arguments and add it to the
defaultProps object

abdelp committed:
Refactor styling

- Add class for wide buttons
- Add prop type validations
- Remove default values from the arguments and add it to the
defaultProps object

abdelp committed:
Merge branch 'feature/styling' of https://github.com/abdelp/react-calculator into feature/styling

abdelp committed:
Refactor default value for result

bellom reviewed (approved):
## Hi @abdelp  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**
